src/main/js/App.js

```diff
@@ -178,7 +209,13 @@ const App = () => {
                     'Content-Type': 'application/json'
                 },
             })
-            .then(res => res.json())
+            .then(res => {
+                if (res.status === 200) {
```

Comment:
Can we generalize this to a function? We seem to reuse this logic a lot, and if we ever change it we don't want to have to change it in several places.